src/main/java/uk/ac/ebi/embl/api/validation/dao/EraproDAOUtilsImpl.java

```diff
@@ -114,14 +114,36 @@ public EraproDAOUtilsImpl (Connection connection)
 	{
 		this.connection=connection;
 	}
-	
+
+
+	private String getAddress(String analysisId) throws SQLException, UnsupportedEncodingException {
+
+		String addressQuery = "select a.center_name, sa.laboratory_name, sa.address, sa.country "
```

Comment:
- For non-brokers, would be safer to get the center_name from the sa.center_name. This would work even if the trigger that sets the analysis.center_name would misbehave somehow.
- For brokers, we need to use sa.broker_name instead of sa.center_name 
- The laboratory name should only be part of address for non-brokers